go/oasis-test-runner/scenario/e2e/runtime/multihost.go

```diff
@@ -0,0 +1,41 @@
+package runtime
+
+import (
+	"github.com/oasisprotocol/oasis-core/go/oasis-test-runner/oasis"
+	"github.com/oasisprotocol/oasis-core/go/oasis-test-runner/scenario"
+)
+
+// MultiHost is the MultiHost node scenario.
+var MultiHost scenario.Scenario = newMultiHostImpl("multihost", "simple-keyvalue-client", nil)
+
+type multiHostImpl struct {
+	runtimeImpl
+}
+
+func newMultiHostImpl(name, clientBinary string, clientArgs []string) scenario.Scenario {
+	return &multiHostImpl{
+		runtimeImpl: *newRuntimeImpl(name, clientBinary, clientArgs),
+	}
+}
+
+func (sc *multiHostImpl) Clone() scenario.Scenario {
+	return &multiHostImpl{
+		runtimeImpl: *sc.runtimeImpl.Clone().(*runtimeImpl),
+	}
+}
+
+func (sc *multiHostImpl) Fixture() (*oasis.NetworkFixture, error) {
+	f, err := sc.runtimeImpl.Fixture()
+	if err != nil {
+		return nil, err
+	}
+
+	// Assign some compute, client and storage workers to shared nodes.
+	f.ComputeWorkers[0].Name = "compute-storage-0"
+	f.StorageWorkers[0].Name = "compute-storage-0"
+
+	f.Clients[0].Name = "client-storage-0"
+	f.StorageWorkers[1].Name = "client-storage-0"
```

Comment:
maybe also test compute-storage-client node (could have two versions of the scenario)